src/java.base/share/classes/java/util/zip/ZipInputStream.java

```diff
@@ -627,6 +628,70 @@ private void readEnd(ZipEntry e) throws IOException {
         }
     }
 
+    /**
+     * Returns true if the ZipEntry has a valid Zip64 extended information extra field.
+     *
+     * This method is used by {@link #readEnd(ZipEntry)} to determine whether the
+     * 'size' and 'compressed size' fields of the Data Descriptor record should
+     * be interpreted as 64-bit numbers instead of the regular 32-bit numbers.
+     *
+     * While ZipOutputStream only produces a Zip64 extra field if either the size or
+     * compressed size of the entry exceeds 0XFFFFFFFF, the ZIP APPNOTE.txt format
+     * specification explicitly allows the use of Zip64 format for entries of any size:
+     *
+     * <pre>
+     * When extracting, if the zip64 extended information extra
+     * field is present for the file the compressed and
+     * uncompressed sizes will be 8 byte values.
+     * </pre>
+     *
+     * To guard against invalid or corrupt extra fields, this method validates that
+     * any Zip64 extended field has a valid block size.
+     *
+     * This method returns false for any invalid extra block sizes, as if the extra
+     * data contained no Zip64 field.
+     */
+    private boolean hasZip64Extra(ZipEntry e) throws IOException {
```

Comment:
Given what this method does, I think it shouldn't throw an IOException (or any other exception for that matter), should strictly return true or false. I haven't imported this code locally, but as far as I can see in its implementation, there doesn't seem to be anything that throws a checked IOException, so perhaps this throws is already redundant?

Reply:
Fixed.